include/net/mqtt_helper.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @defgroup mqtt_helper MQTT helper library
+ * @{
+ * @brief Convenience library that simplifies Zephyr MQTT API and socket handling.
+ */
+
+#ifndef MQTT_HELPER__
+#define MQTT_HELPER__
+
+#include <stdio.h>
+#include <zephyr/net/mqtt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum mqtt_state {
+	MQTT_STATE_UNINIT,
+	MQTT_STATE_DISCONNECTED,
+	MQTT_STATE_TRANSPORT_CONNECTING,
+	MQTT_STATE_CONNECTING,
+	MQTT_STATE_TRANSPORT_CONNECTED,
+	MQTT_STATE_CONNECTED,
+	MQTT_STATE_DISCONNECTING,
+
+	MQTT_STATE_COUNT,
+};
+
+enum mqtt_helper_error {
+	/** The received payload is larger than the payload buffer. */
+	MQTT_HELPER_ERROR_MSG_SIZE,
+};
+
+struct mqtt_helper_buf {
+	/** Pointer to buffer. */
+	char *ptr;
+
+	/** Size of buffer. */
+	size_t size;
+};
+
+typedef void (*mqtt_helper_handler_t)(struct mqtt_evt *evt);
+typedef void (*mqtt_helper_on_connack_t)(enum mqtt_conn_return_code return_code);
+typedef void (*mqtt_helper_on_disconnect_t)(int result);
+typedef void (*mqtt_helper_on_publish_t)(struct mqtt_helper_buf topic_buf,
+					 struct mqtt_helper_buf payload_buf);
+typedef void (*mqtt_helper_on_puback_t)(uint16_t message_id, int result);
+typedef void (*mqtt_helper_on_suback_t)(uint16_t message_id, int result);
+typedef void (*mqtt_helper_on_pingresp_t)(void);
+typedef void (*mqtt_helper_on_error_t)(enum mqtt_helper_error error);
+
+struct mqtt_helper_cfg {
```

Comment:
This and the function pointers probably need documentation

Reply:
Jepp, will extend documentation for the library in a follow up PR. Its quite minuscule atm

Reply:
I'll also expand mqtt_helper.rst

Reply:
Sounds good